### PR TITLE
Add direct robot assignment api call

### PR DESCRIPTION
**Pull request by cnboonhan**
This PR aims to add direct robot assignment REST api calls in the rmf_server, following this: https://github.com/open-rmf/rmf_demos/blob/redesign_v2/rmf_demos_tasks/rmf_demos_tasks/direct_patrol.py

### CODE REVIEW

**Review by koonpeng — CHANGES_REQUESTED** [review]
Can you add some tests? See the dispatch task tests for examples.

File: packages/api-server/api_server/routes/tasks/tasks.py [hunk 1]
```diff
@@ -127,6 +127,26 @@ async def post_dispatch_task(
...
+    request: mdl.RobotTaskRequest = Body(...),
+    task_repo: TaskRepository = Depends(task_repo_dep),
+):
+    resp = mdl.RobotTaskResponse.parse_raw(
+        await tasks_service.call(request.json(exclude_none=True))
+    )
+    if not resp.__root__.__root__.success:
+        return RawJSONResponse(resp.json(), 400)
+    await task_repo.save_task_state(
+        cast(mdl.TaskDispatchResponse, resp.__root__).__root__.state
```
Comment (koonpeng) [inline packages/api-server/api_server/routes/tasks/tasks.py:145]: This is giving type errors, do you perhaps meant to 

```suggestion
        cast(mdl.TaskDispatchResponseItem, resp.__root__).__root__.state
```

Comment (cnboonhan) [reply packages/api-server/api_server/routes/tasks/tasks.py]: tried as you described, but i get another error, `__root__` is unknown, in the api-server tests. I'm not exactly sure on this.

**Comment by youliangtan** [comment]
Test the  `/robot_task` endpoint with the json request body below:  (rmf_demos's office world)
```
{
  "type": "robot_task_request",
  "robot": "tinyRobot1",
  "fleet": "tinyRobot",
  "request": {
    "unix_millis_earliest_start_time": 0,
    "category": "patrol",
    "description": {
      "places": [
        "pantry",
        "lounge"
      ],
      "rounds": 2
    }
  }
}
```

**Review by koonpeng — APPROVED** [review]

**Pull request merged**